### PR TITLE
Add yaml serialization tags to breaker config

### PR DESCRIPTION
**Pull request by marcoferrer**
This change allows introduces yaml tags to the fields of the circuit breaker config. Allowing config to be sourced from a common file and removing the need for a code change when making adjustments

### CODE REVIEW

File: breakerbp/failure_ratio.go
Comment (fishy) [inline breakerbp/failure_ratio.go:29]: why is this "primary"?

**Review by fishy — CHANGES_REQUESTED** [review]
Note that this is technically a breaking change, as if someone is already using it with yaml, they need to change the keys of their yaml config for their code to continue to work, so after approved we'll only merge this after we are done with all the non-breaking 0.8.x changes.

File: breakerbp/failure_ratio.go
Comment (konradreiche) [reply breakerbp/failure_ratio.go:29]: Ideally this can be mapped to the actual field name, e.g.
```go
`yaml:"minRequestsToTrip"`
```

Comment (marcoferrer) [reply breakerbp/failure_ratio.go:29]: Thats was a miss. I agree `yaml:"minRequestsToTrip"` is the correct value.
- [ ] Fix min requests tag

**Comment by marcoferrer** [comment]
@fishy Would changing the keys to pascal case make this change backwards compatible with any existing serialization usages?

**Comment by fishy** [comment]
@marcoferrer I have lost track of what "pascal case" is 😅 , but the only true backward compatible way is to use all lower case (e.g. `minrequeststotrip` for field `MinRequestsToTrip`), which is the same as don't specify yaml tag at all.

also I'm not saying we should not make breaking changes. I'm just saying we need to slightly delay the merging of approved breaking changes as we don't want to maintain a `0.8.x` branch :)

**Review by fishy — APPROVED** [review]

**Review by pacejackson — APPROVED** [review]

**Review by konradreiche — APPROVED** [review]

**Pull request merged**
